builtin/v9/migration/verifreg.go

```diff
@@ -0,0 +1,143 @@
+)
+
+func migrateVerifreg(ctx context.Context, adtStore adt8.Store, priorEpoch abi.ChainEpoch, initStateV9 init9.State, marketStateV8 market8.State, verifregStateV8 verifreg8.State, emptyMapCid cid.Cid) (cid.Cid, map[abi.DealID]verifreg9.AllocationId, error) {
+	pendingProposals, err := adt8.AsSet(adtStore, marketStateV8.PendingProposals, builtin.DefaultHamtBitwidth)
+	if err != nil {
+		return cid.Undef, nil, xerrors.Errorf("failed to load pending proposals: %w", err)
+	}
+
+	proposals, err := market8.AsDealProposalArray(adtStore, marketStateV8.Proposals)
+	if err != nil {
+		return cid.Undef, nil, xerrors.Errorf("failed to get proposals: %w", err)
+	}
+
+	pendingMap := make(map[abi.DealID]market8.DealProposal)
+	var proposal market8.DealProposal
+	if err = proposals.ForEach(&proposal, func(dealID int64) error {
+		pcid, err := proposal.Cid()
+		if err != nil {
+			return err
+		}
+
+		isPending, err := pendingProposals.Has(abi.CidKey(pcid))
+		if err != nil {
+			return xerrors.Errorf("failed to check pending: %w", err)
+		}
+
+		if isPending {
+			pendingMap[abi.DealID(dealID)] = proposal
+		}
+
+		return nil
+	}); err != nil {
+		return cid.Undef, nil, xerrors.Errorf("failed to iterate over proposals: %w", err)
+	}
+
+	nextAllocationID := verifreg9.AllocationId(1)
+	allocationsMapMap := make(map[address.Address]*adt9.Map)
+	dealsToAllocations := make(map[abi.DealID]verifreg9.AllocationId)
+	for dealID, proposal := range pendingMap {
+		clientIDAddress, ok, err := initStateV9.ResolveAddress(adtStore, proposal.Client)
+		if err != nil {
+			return cid.Undef, nil, xerrors.Errorf("failed to resolve client %s: %w", proposal.Client, err)
+		}
+
+		if !ok {
+			return cid.Undef, nil, xerrors.New("failed to find client in init actor map")
+		}
+
+		clientIDu64, err := address.IDFromAddress(clientIDAddress)
+		if err != nil {
+			return cid.Undef, nil, err
+		}
+
+		providerIDAddress, ok, err := initStateV9.ResolveAddress(adtStore, proposal.Provider)
+		if err != nil {
+			return cid.Undef, nil, xerrors.Errorf("failed to resolve provider %s: %w", proposal.Provider, err)
+		}
+
+		if !ok {
+			return cid.Undef, nil, xerrors.New("failed to find provider in init actor map")
+		}
+
+		providerIDu64, err := address.IDFromAddress(providerIDAddress)
+		if err != nil {
+			return cid.Undef, nil, err
+		}
+
+		clientAllocationMap, ok := allocationsMapMap[clientIDAddress]
+		if !ok {
+			clientAllocationMap, err = adt9.AsMap(adtStore, emptyMapCid, builtin.DefaultHamtBitwidth)
+			if err != nil {
+				return cid.Undef, nil, xerrors.Errorf("failed to load empty map: %w", err)
+			}
+
+			allocationsMapMap[clientIDAddress] = clientAllocationMap
+		}
+
+		if err = clientAllocationMap.Put(nextAllocationID, &verifreg9.Allocation{
+			Client:     abi.ActorID(clientIDu64),
+			Provider:   abi.ActorID(providerIDu64),
+			Data:       proposal.PieceCID,
+			Size:       proposal.PieceSize,
+			TermMin:    proposal.Duration(),
+			TermMax:    market9.DealMaxDuration,
```

Comment:
‼️ The max term should be a policy value _more than_ the deal duration, not a fixed value for all deals.

Reply:
@anorth I based this on the [text of the FIP ](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0045.md#migration). Am I misunderstanding something here?

Reply:
You're right, I take it back.

Reply:
Zen had a slightly different suggestion [here](https://github.com/filecoin-project/go-state-types/pull/85#discussion_r988251213) -- do you have an opinion?

Reply:
Wait, yeah I un-take it back. We want the same behaviour as the market would have done. If a client set 540d deal term, the max should be +buffer on that.

Reply:
Good, thanks! I will implement this change and update the FIP for overall Zen.